sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -996,10 +996,34 @@ def test_init_param():
     gmm.fit(X)
     default_means = gmm.means_
 
-    INIT_TYPE = ['random', 'rand_data', 'k-means++']
+    INIT_TYPE = ['random', 'rand_data', 'k-means++', 'kmeans']
 
     for init in INIT_TYPE:
         gmm = GaussianMixture(n_components=n_components, random_state=rng,
                               init_params=init, tol=1e-06, max_iter=1000)
         gmm.fit(X)
         assert_almost_equal(default_means, gmm.means_, decimal=1)
+
+    # Check that all initialisations provide unique starting means
+
+    rand_data = RandomData(rng, scale=5)
+    X = rand_data.X['full']
+
+    gmm = GaussianMixture(n_components=n_components, random_state=rng,
+                          init_params='kmeans', max_iter=1)
+    gmm.fit(X)
+    default_means_init = {}
+
+    for init in INIT_TYPE:
+        gmm = GaussianMixture(n_components=n_components, random_state=rng,
```

Comment:
You need to use random_state=0 or some other int, or variation will be due to rng